packages/back-end/src/controllers/features.ts

```diff
@@ -1656,26 +1749,61 @@ export async function putFeatureRule(
   ) {
     context.permissions.throwPermissionError();
   }
+  let canUpdateRevision = true;
+  if (rule.type === "safe-rollout") {
+    if (!rule.safeRolloutId) {
+      throw new Error("Safe rollout rule must have a safeRolloutId");
+    }
+    const existingSafeRollout = await context.models.safeRollout.getById(
+      rule.safeRolloutId
+    );
+    if (!existingSafeRollout) {
+      throw new Error("Safe rollout rule must have a safeRolloutId");
+    }
 
-  const revision = await getDraftRevision(context, feature, parseInt(version));
-  const resetReview = resetReviewOnChange({
-    feature,
-    changedEnvironments: [environment],
-    defaultValueChanged: false,
-    settings: org?.settings,
-  });
-  await editFeatureRule(
-    revision,
-    environment,
-    i,
-    rule,
-    res.locals.eventAudit,
-    resetReview
-  );
+    if (interfaceFields) {
+      await context.models.safeRollout.update(existingSafeRollout, {
+        ...omit(interfaceFields, [
+          "organization",
+          "dateCreated",
+          "dateUpdated",
+          "startedAt",
+        ]),
+      });
+    }
+    if (existingSafeRollout.startedAt) {
+      // not sure if we want to lock or just create a new revision?
+      // revsion is locked for the safe rollout once it's released
+      canUpdateRevision = false;
```

Comment:
I don't think we should block this, but we should have checks to make sure only allowed fields are changed.

Some fields like `trackingKey` we never want to allow updates to. Others like `hashAttribute` are fine before it's started, but should be blocked after.